doc/packages/build-support.md

```diff
@@ -0,0 +1,92 @@
+# Build Support {#sec-build-support}
+
+## `pkgs.substitute` {#pkgs-substitute}
+
+`pkgs.substitute` is a wrapper around [the `substitute` function](#fun-substitute) provided by the
+stdenv. It replaces strings in `src` as specified by the `substitutions` argument.
+
+Whereas in a build script, one might call `substitute $infile $outfile --replace-fail @foo@
+${foopkg}/bin/foo`, one might accomplish this with `pkgs.substitute` as such:
+
+```nix
+{ substitute, foopkg }:
+substitute {
+  src = ./sourcefile.txt;
+  substitutions = [
+    "--replace"
+    "@foo@"
+    "${foopkg}/bin/foo"
+  ];
+}
+```
+
+## `pkgs.substituteAll` {#pkgs-substituteall}
+
+`pkgs.substituteAll` is a wrapper around [the `substituteAll` function](#fun-substituteAll) provided
+by the stdenv. It substitutes all instances of `@varName@` in `src` with the value of environment
+variable `varName`. As this uses the `substituteAll` function, its limitations also apply here,
```

Comment:
```suggestion
variable `varName`. As this uses the `substituteAll` function, it has the following limitations:
```

Reply:
Also: isn't that already documented in stdenv `substituteAll` ? We should only maintain those in one place.

Reply:
shall we link to that section then?

Reply:
I'm unclear if you're recommending that I copy the limitations from above, or if that I should remove them because they're already a duplication of existing documentation. I may have misunderstood, but the prior reviewer seems to indicate that I should include them there: https://github.com/NixOS/nixpkgs/pull/326383#discussion_r1677007698 -- would a link to the existing function be your preference?

Reply:
> would a link to the existing function be your preference

That one i think.

Limitations see [link](...)